packages/react-server/src/ReactFizzServer.js

```diff
@@ -303,9 +312,11 @@ export function createRequest(
 }
 
 function pingTask(request: Request, task: Task): void {
-  const pingedTasks = request.pingedTasks;
-  pingedTasks.push(task);
-  if (pingedTasks.length === 1) {
+  const priority = task.priority;
+  const queue =
+    priority === IDLE_PRIORITY ? request.idleTasks : request.pingedTasks;
+  queue.push(task);
+  if (queue.length === 1) {
```

Comment:
the real heuristic here should "is the length of all queues === 1" but if we end up with many queues that logic gets cumbersome. Really want we want is to only `scheduleWork` if it's needed and this callsite can't exclusively know that. Extra calls aren't that bad b/c they won't do anything if there is no work to perform but it's a waste of compute.

would love thoughts on this one